datafusion/physical-expr/src/regex_expressions.rs

```diff
@@ -332,10 +482,70 @@ pub fn specialize_regexp_replace<T: OffsetSizeTrait>(
 
 #[cfg(test)]
 mod tests {
-    use super::*;
     use arrow::array::*;
+
     use datafusion_common::ScalarValue;
 
+    use super::*;
+
+    #[test]
+    fn test_case_sensitive_regexp_like() {
+        let values = StringArray::from(vec!["abc"; 5]);
+
+        let patterns =
+            StringArray::from(vec!["^(a)", "^(A)", "(b|d)", "(B|D)", "^(b|c)"]);
+
+        let mut expected_builder: BooleanBuilder = BooleanBuilder::new();
+        expected_builder.append_value(true);
+        expected_builder.append_value(false);
+        expected_builder.append_value(true);
+        expected_builder.append_value(false);
+        expected_builder.append_value(false);
+        let expected = expected_builder.finish();
```

Comment:
This probably just follows the exisitng code in this file, but it is a bit more concise to build `BooleanArry`s like:

Something like this (untested):
```suggestion
        let expected = BooleanArray::from(vec![true, false, true, false, false]);
```

Reply:
This is just me copy/pasting from some other bit of code and not thinking too much about it. I am still getting used to Rust's way of doing things :)

Reply:
I think following the existing structure is fine, I just figured I would point it out